### PR TITLE
[RISCV] Split and rename WriteVISlideX into WriteVSlideUpX and WriteVSlideDownX

### PR DESCRIPTION
**Pull request by mshockwave**
Some processors might have different latencies and/or rthroughput for slide up and down operations on integer vectors, yet there is only a single SchedWrite for both of them at this moment. This patch splits this SchedWrite into two as well as drop the "I" before "Slide" since such information is redundant. We also do the same renaming on `WriteVISlideI`.
Note that we only split the X variant (i.e. using a register value for index offset) for now.

This is effectively NFC.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVInstrInfoV.td [hunk 1]
```diff
@@ -976,8 +976,12 @@ multiclass VNCLP_IV_V_X_I<string opcodestr, bits<6> funct6> {
 }
 
 multiclass VSLD_IV_X_I<string opcodestr, bits<6> funct6> {
+  // Note: In the future, if VISlideI is also splitted into VISlideUpI and
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVInstrInfoV.td:979]: splitted -> split

File: llvm/lib/Target/RISCV/RISCVScheduleV.td [hunk 2]
```diff
@@ -514,7 +514,8 @@ def WriteVMovXS : SchedWrite;
 def WriteVMovSF : SchedWrite;
 def WriteVMovFS : SchedWrite;
 // 16.3. Vector Slide Instructions
-defm "" : LMULSchedWrites<"WriteVISlideX">;
+defm "" : LMULSchedWrites<"WriteVISlideUpX">;
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVScheduleV.td:517]: Can we drop the I before Slide? These aren't restricted to integer vectors. The scalar operand for the is an integer but that's already covered by I and X at the end. And there's no separate F version.

Comment (mshockwave) [reply llvm/lib/Target/RISCV/RISCVScheduleV.td]: Done. I also renamed `VISlideI` to `VSlideI`.

File: llvm/lib/Target/RISCV/RISCVInstrInfoV.td (hunk 1 above)
Comment (mshockwave) [reply llvm/lib/Target/RISCV/RISCVInstrInfoV.td]: Fixed.

File: llvm/lib/Target/RISCV/RISCVInstrInfoV.td [hunk 3]
```diff
@@ -976,10 +976,14 @@ multiclass VNCLP_IV_V_X_I<string opcodestr, bits<6> funct6> {
 }
 
 multiclass VSLD_IV_X_I<string opcodestr, bits<6> funct6> {
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVInstrInfoV.td:978]: Can we pass a bit as an argument to indicate up or down? That seems like a better interface then making assumptions about the name.

File: llvm/lib/Target/RISCV/RISCVSchedSiFive7.td [hunk 4]
```diff
@@ -937,8 +937,9 @@ foreach mx = SchedMxList in {
   defvar Cycles = SiFive7GetCyclesDefault<mx>.c;
   defvar IsWorstCase = SiFive7IsWorstCaseMX<mx, SchedMxList>.c;
   let Latency = 4, AcquireAtCycles = [0, 1], ReleaseAtCycles = [1, !add(1, Cycles)] in {
-    defm "" : LMULWriteResMX<"WriteVISlideX",   [SiFive7VCQ, SiFive7VA], mx, IsWorstCase>;
```
Comment (wangpc-pp) [inline llvm/lib/Target/RISCV/RISCVSchedSiFive7.td:940]: The spaces between `,` and `[` mismatched, should we make them consistent?

File: llvm/lib/Target/RISCV/RISCVInstrInfoV.td (hunk 3 above)
Comment (mshockwave) [reply llvm/lib/Target/RISCV/RISCVInstrInfoV.td]: I agree. It's fixed now.

File: llvm/lib/Target/RISCV/RISCVSchedSiFive7.td (hunk 4 above)
Comment (mshockwave) [reply llvm/lib/Target/RISCV/RISCVSchedSiFive7.td:940]: It's fixed now.

**Review by topperc — APPROVED** [review]
LGTM

File: llvm/lib/Target/RISCV/RISCVInstrInfoVPseudos.td [hunk 5]
```diff
@@ -6861,8 +6863,8 @@ let mayLoad = 0, mayStore = 0, hasSideEffects = 0 in {
 // 16.3. Vector Slide Instructions
 //===----------------------------------------------------------------------===//
 let Predicates = [HasVInstructions] in {
-  defm PseudoVSLIDEUP    : VPseudoVSLD_VX_VI<uimm5, "@earlyclobber $rd">;
-  defm PseudoVSLIDEDOWN  : VPseudoVSLD_VX_VI<uimm5>;
+  defm PseudoVSLIDEUP    : VPseudoVSLD_VX_VI<uimm5, /*slidesUp=*/true, "@earlyclobber $rd">;
```
Comment (wangpc-pp) [inline llvm/lib/Target/RISCV/RISCVInstrInfoVPseudos.td:6866]: You can just use `slidesUp=true` here.
https://llvm.org/docs/TableGen/ProgRef.html#record-bodies

**Pull request merged**